src/engine/Story.ts

```diff
@@ -1830,7 +1964,7 @@ export class Story extends InkObject {
   }
 
   public RemoveVariableObserver(
-    observer: Story.VariableObserver,
+    observer: Story.VariableObserver | null = null,
     specificVariableName: string
```

Comment:
Not sure this is part of the logic change, but I'll comment before forget: specificVariableName also defaults to null upstream.

Reply:
Ah yes so it seems normal, but we can possibly type it as `specificVariableName?: string` to make it explicitly optional?

Reply:
I'd say… we can't a the moment, because we originally decided to closely match the C# "types", so `undefined` is avoided as much as possible.

But maybe we should reconsider now, and when it makes sense, use `undefined` instead of `null` to really highlight the optional nature of a parameter / property.

Reply:
As for specificVariableName, I think what happened is that this code was originally written by another contributor and that rule isn't explicit.

Bu now that I think of it, it makes a of sense to stick to undefined for methods exposed outside of the runtime. Otherwise it's going to be confusing. I'll update that code and touch up the documentation at the same time.